### PR TITLE
Adds hparam filterbar to runs table

### PR DESCRIPTION
**Pull request by hoonji**
## Motivation for features / changes
To view applied hparam filters in the runs table.

## Technical description of changes
If a user filters the runs table based on a hparam column, the filter will be shown as a chip above the table. Clicking the chip will allow the user to modify the filter via a dialog. Clicking the close button on the chip will remove the filter.

## Screenshots of UI changes (or N/A)
Screenshot 2023-12-04 at 18 30 27

### CODE REVIEW

File: tensorboard/webapp/runs/views/runs_table/runs_table_container.ts [hunk 1]
```diff
@@ -349,7 +349,7 @@ export class RunsTableContainer implements OnInit, OnDestroy {
   addHparamFilter(event: FilterAddedEvent) {
     this.store.dispatch(
       hparamsActions.dashboardHparamFilterAdded({
-        name: event.header.name,
```
Comment (hoonji) [inline tensorboard/webapp/runs/views/runs_table/runs_table_container.ts:352]: On inspection, it appears filter actions can be independent of "Headers". We just use the name field of the headers anyway.

**Comment by hoonji** [comment]
The cl requires some minor changes (need to add icon once I get tf_mirror permissions), but the other parts are ready for review. This is my first major PR on this code base - please be as critical as necessary!

File: tensorboard/webapp/runs/views/runs_table/runs_table_container.ts (hunk 1 above)
Comment (JamesHollyer) [reply tensorboard/webapp/runs/views/runs_table/runs_table_container.ts:352]: I probably would have made a quick PR with just this as a standalone change. This is fine though.

**Review by JamesHollyer — COMMENTED** [review]
This all looks good. I am only not approving as I want to see the finished PR with the correct icon.

File: tensorboard/webapp/runs/views/runs_table/runs_table_container.ts (hunk 1 above)
Comment (hoonji) [reply tensorboard/webapp/runs/views/runs_table/runs_table_container.ts:352]: Ack - I'll split these sorts of changes going forward

File: tensorboard/webapp/runs/views/runs_table/hparam_filterbar_component.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import {RangeValues} from '../../../widgets/range_input/types';
+import {CustomModalComponent} from '../../../widgets/custom_modal/custom_modal_component';
+
+@Component({
+  selector: 'hparam-filterbar-component',
+  templateUrl: 'hparam_filterbar_component.ng.html',
+  styleUrls: ['hparam_filterbar_component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HparamFilterbarComponent {
```
[the remark this answers is not in the document]
Comment (hoonji) [reply tensorboard/webapp/runs/views/runs_table/hparam_filterbar_component.ts]: Agreed that it should be useful for other filters. Done.

**Comment by hoonji** [comment]
All changes applied (including the icon)! PTAL.

**Review by JamesHollyer — APPROVED** [review]
LGTM.

Please re-request the review when appropriate.
Screenshot 2023-12-06 at 10 30 40 AM

**Comment by hoonji** [comment]
Right, I was actually looking for something like that. Thanks!

**Pull request merged**